source/dofs/dof_handler.cc

```diff
@@ -2006,10 +2006,7 @@ DoFHandler<dim, spacedim>::~DoFHandler()
   this->tria_listeners.clear();
 
   // release allocated memory
-  // virtual functions called in constructors and destructors never use the
-  // override in a derived class
-  // for clarity be explicit on which function is called
-  DoFHandler<dim, spacedim>::clear();
+  clear();
```

Comment:
I guess you're changing this because we no longer have classes derived from `DoFHandler`?

Reply:
Exactly. We can also get rid of most `this` pointers in the implementation.

However, this does not have to be part of this PR. I tend to move it to a separate one.

Reply:
That's ok, but it's also fine as is.

Reply:
Hmm, we still have the deprecated `hp::DoFHandler` around that will probably need this. I'll revert it.

Reply:
It really doesn't make a difference in practice. The code is just clearly telling what's happening here (which some static analyzers demanded).